platform-intel.c

```diff
@@ -669,14 +733,274 @@ static int read_efi_variable(void *buffer, ssize_t buf_size,
+	}
+	return MDADM_STATUS_SUCCESS;
+}
+
+/**
+ * read_uefi_acpi_orom_data() - read OROM data from UEFI ACPI table.
+ * @fd: file descriptor.
+ * @file_name: name of ACPI table.
+ * @uefi_table: struct to fill out.
+ *
+ * Read OROM from ACPI UEFI table under given file descriptor.
+ * Table must have the appropriate OROM data, which should be confirmed before call this function.
+ * In case of success, &orom in structure in &uefi_table will be filled..
+ *
+ * Return: %MDADM_STATUS_SUCCESS on success, %MDADM_STATUS_ERROR otherwise.
+ */
+mdadm_status_t
+read_uefi_acpi_orom_data(int fd, char *file_name, uefi_acpi_table_with_orom_t *uefi_table)
+{
+	assert(is_fd_valid(fd));
+
+	// set data offset for OROM read
+	if (lseek(fd, uefi_table->table.dataOffset, 0) == -1L) {
+		pr_err("Fail to reposition offset in ACPI UEFI file. File: %s, Error: %s\n",
+		       file_name, strerror(errno));
+		return MDADM_STATUS_ERROR;
+	}
+
+	if (read(fd, &uefi_table->orom, sizeof(uefi_table->orom)) == -1) {
+		pr_err("Fail to read OROM data from ACPI UEFI file. File: %s, Error: %s\n",
+		       file_name, strerror(errno));
+		return MDADM_STATUS_ERROR;
+	}
+
+	return MDADM_STATUS_SUCCESS;
+}
+
+/**
+ * verify_uefi_acpi_table_length() - verify if ACPI UEFI table have correct length with focus at
+ * OROM.
+ * @table: struct with UEFI table.
+ *
+ * Verify if ACPI UEFI table have correct length with focus at OROM. Make sure that the file is
+ * correct and contains the appropriate length data based on the length of the OROM.
+ *
+ * Return: %true if length is correct, %false otherwise.
+ */
+bool verify_uefi_acpi_table_length(struct uefi_acpi_table table)
+{
+	if (table.length < ACPI_UEFI_DATA_OFFSET)
+		return false;
+
+	if (table.length <= table.dataOffset)
+		return false;
+
+	if (table.length - table.dataOffset != sizeof(struct imsm_orom))
+		return false;
+	return true;
+}
+
+/**
+ * find_orom_in_acpi_uefi_tables() - find OROM in UEFI ACPI tables based on requested OROM ids.
+ * @imsm_orom_ids: array of IMSM OROM Identifiers.
+ * @imsm_orom_ids_number: number of IMSM OROM Identifiers.
+ * @orom: OROM struct buffer to fill out.
+ *
+ * Find OROM in UEFI ACPI tables provided by Intel, based on requested controllers.
+ * The first one to be matched, will be used.
+ * If found, the buffer with the OROM structure will be filled.
+ *
+ * Return: %MDADM_STATUS_SUCCESS on success, %MDADM_STATUS_ERROR otherwise.
+ */
+mdadm_status_t
+find_orom_in_acpi_uefi_tables(imsm_orom_id_t *imsm_orom_ids, int imsm_orom_ids_number,
+			      struct imsm_orom *orom)
+{
+	mdadm_status_t status = MDADM_STATUS_ERROR;
+	uefi_acpi_table_with_orom_t uefi_table;
+	char path[PATH_MAX];
+	struct dirent *ent;
+	int fd = -1;
+	DIR *dir;
+
+	dir = opendir(ACPI_TABLES_PATH);
+	if (!dir)
+		return MDADM_STATUS_ERROR;
+
+	for (ent = readdir(dir); ent; ent = readdir(dir)) {
+		close_fd(&fd);
+
+		/* Check if file is a UEFI table */
+		if (strncmp(ent->d_name, ACPI_UEFI_TABLE_BASE_NAME,
```

Comment:
What if there will be UEFI1/2/3 instead of UEFI?
-r-------- 1 root root     66 Aug  2 09:16 UEFI1
-r-------- 1 root root     92 Aug  2 09:16 UEFI2
-r-------- 1 root root     92 Aug  2 09:16 UEFI3
-r-------- 1 root root     92 Aug  2 09:16 UEFI4

Reply:
This check accepted all files starting with UEFI

Reply:
Ack, didn't see != 0 and was thinking == 0 is used :)